omegaconf/omegaconf.py

```diff
@@ -901,8 +902,10 @@ def _select_one(
     assert isinstance(c, (DictConfig, ListConfig)), f"Unexpected type : {c}"
     if isinstance(c, DictConfig):
         assert isinstance(ret_key, str)
-        val: Optional[Node] = c._get_node(ret_key, validate_access=False)
+        val = c._get_node(ret_key, validate_access=False)
+        assert val is None or isinstance(val, Node)
         if val is not None:
+            assert isinstance(val, Node)
```

Comment:
```suggestion
        assert val is None or isinstance(val, Node)
        if val is not None:
```

Technically the second assertion here is not necessary: mypy can infer that `val` has type `Node` based on the previous two lines.

Reply:
(If you type `reveal_type(val)` inside of the `if` block, mypy prints `Revealed type is 'omegaconf.base.Node'`)